src/pal/src/map/virtual.cpp

```diff
@@ -826,10 +832,8 @@ static LPVOID VIRTUALResetMemory(
 
     TRACE( "Resetting the memory now..\n");
 
-    StartBoundary = (UINT_PTR)lpAddress & ~VIRTUAL_PAGE_MASK;
-    // Add the sizes, and round down to the nearest page boundary. 
-    MemSize = ( ((UINT_PTR)lpAddress + dwSize + VIRTUAL_PAGE_MASK) & ~VIRTUAL_PAGE_MASK ) - 
-               StartBoundary;
+    StartBoundary = (UINT_PTR) ALIGN_DOWN(lpAddress, GetVirtualPageSize());
+    MemSize = ALIGN_UP((UINT_PTR)lpAddress + dwSize, GetVirtualPageSize()) - StartBoundary;
```

Comment:
@janvorli Yes you are right.  I had been reading the first VIRTUAL_PAGE_MASK as VIRTUAL_PAGE_SIZE.  Sometimes you are so blind, you cannot see something even if someone shows you.  Thanks for insisting..